bundles/org.palladiosimulator.textual.tpcm/src/org/palladiosimulator/textual/tpcm/scoping/TPCMImportUriGlobalScopeProvider.java

```diff
@@ -0,0 +1,49 @@
+package org.palladiosimulator.textual.tpcm.scoping;
+
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+
+import javax.inject.Inject;
+
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.xtext.EcoreUtil2;
+import org.eclipse.xtext.scoping.impl.ImportUriGlobalScopeProvider;
+import org.eclipse.xtext.util.IAcceptor;
+import org.eclipse.xtext.util.IResourceScopeCache;
+import org.palladiosimulator.textual.tpcm.language.Import;
+import org.palladiosimulator.textual.tpcm.language.Model;
+
+import com.google.common.collect.Iterators;
+import com.google.inject.Provider;
+
+public class TPCMImportUriGlobalScopeProvider extends ImportUriGlobalScopeProvider {
+	@Inject
+	IResourceScopeCache cache;
+
+	@Override
+	protected LinkedHashSet<URI> getImportedUris(Resource resource) {
```

Comment:
nasty types but derived from xtext

Reply:
This is taken from an blog post of an xtext developer team member.